front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java

```diff
@@ -469,17 +482,58 @@ public Void call() throws Exception {
         return objectMapper.readValue(json, clas);
       } catch (Exception ex) {
         if (current_version) {
-          log.error("Error reading {}: {}", value("object", object.getName()), ex);
+            log.error("Error reading {}: {}", value("object", object.getName()), ex.getMessage());
         } else {
           log.error("Error reading {} generation={}: {}",
                     value("object", object.getName()),
                     value("generation", object.getGeneration()),
-                    ex);
+                    ex.getMessage());
         }
         return null;
     }
   }
 
+  private void scheduleWriteLastModified(String daoTypeName) {
+    long modifiedTime = getLastModifiedOfTypeName(daoTypeName);
+    Date when = new Date();
+
+    synchronized (daoTypeNameReferenceTime) {
+      boolean schedule = daoTypeNameReferenceTime.get(daoTypeName) == null;
+      daoTypeNameReferenceTime.put(daoTypeName, modifiedTime);
+      GcsStorageService service = this;
+
+      if (schedule) {
+          Runnable task = new Runnable() {
+          public void run() {
+             log.info("RUNNING {}", daoTypeName);
+             Long storedTime;
+             synchronized (service.daoTypeNameReferenceTime) {
+                 storedTime = service.daoTypeNameReferenceTime.remove(daoTypeName);
+                 if (storedTime == null) {
+                   // unexpected
```

Comment:
Reading this i'm not sure how this can happen, is that what you mean by "unexpected"?